windows/CodePushNativeModule.cs

```diff
@@ -0,0 +1,291 @@
+                }
+
+                if (updateState == (int)UpdateState.Pending && !currentUpdateIsPending)
+                {
+                    // The caller wanted a pending update
+                    // but there isn't currently one.
+                    promise.Resolve("");
+                }
+                else if (updateState == (int)UpdateState.Running && currentUpdateIsPending)
+                {
+                    // The caller wants the running update, but the current
+                    // one is pending, so we need to grab the previous.
+                    promise.Resolve(await _codePush.UpdateManager.GetPreviousPackage());
+                }
+                else
+                {
+                    // The current package satisfies the request:
+                    // 1) Caller wanted a pending, and there is a pending update
+                    // 2) Caller wanted the running update, and there isn't a pending
+                    // 3) Caller wants the latest update, regardless if it's pending or not
+                    if (_codePush.IsRunningBinaryVersion)
+                    {
+                        // This only matters in Debug builds. Since we do not clear "outdated" updates,
+                        // we need to indicate to the JS side that somehow we have a current update on
+                        // disk that is not actually running.
+                        currentPackage["_isDebugOnly"] = true;
+                    }
+
+                    // Enable differentiating pending vs. non-pending updates
+                    currentPackage["isPending"] = currentUpdateIsPending;
+                    promise.Resolve(currentPackage);
+                }
+            };
+
+            Context.RunOnNativeModulesQueueThread(getCurrentPackageAction);
+        }
+
+
+        [ReactMethod]
+        public void getNewStatusReport(IPromise promise)
+        {
+            // TODO implement this
+            promise.Resolve("");
+        }
+
+        [ReactMethod]
+        public void installUpdate(JObject updatePackage, int installMode, int minimumBackgroundDuration, IPromise promise)
+        {
+            Action installUpdateAction = async () =>
+            {
+                await _codePush.UpdateManager.InstallPackage(updatePackage, SettingsManager.IsPendingUpdate(null));
+                var pendingHash = (string)updatePackage[CodePushConstants.PackageHashKey];
+                SettingsManager.SavePendingUpdate(pendingHash, /* isLoading */false);
+                if (installMode == (int)InstallMode.OnNextResume)
+                {
+                    if (_codePushLifecycleEventListener == null)
+                    {
+                        // Ensure we do not add the listener twice.
+                        Action loadBundleAction = () =>
+                        {
+                            Context.RunOnNativeModulesQueueThread(async () =>
```

Comment:
Does this need to be dispatched to the native modules thread? Since the reload logic just immediately dispatches back to the UI thread, and this operation doesn't need to actually be async, I don't think we need to worry about doing this unless it's required.